src/ReactiveUI/ViewLocator.cs

```diff
@@ -93,31 +95,27 @@ public DefaultViewLocator(Func<string, string> viewModelToViewFunc = null)
         public IViewFor ResolveView<T>(T viewModel, string contract = null)
             where T : class
         {
-            var view = this.AttemptViewResolutionFor(viewModel.GetType(), contract);
+            var view = GetCandidateViewModelTypes(viewModel)
```

Comment:
Benchmark results

```
Host Process Environment Information:
BenchmarkDotNet.Core=v0.9.9.0
OS=Microsoft Windows NT 6.2.9200.0
Processor=Intel(R) Xeon(R) CPU E3-1231 v3 3.40GHz, ProcessorCount=8
Frequency=3320309 ticks, Resolution=301.1768 ns, Timer=TSC
CLR=MS.NET 4.0.30319.42000, Arch=32-bit RELEASE
GC=Concurrent Workstation
JitModules=clrjit-v4.7.2046.0

Type=ViewResolutionBenchmark  Mode=Throughput

                Method |    Median |    StdDev | Scaled | Scaled-SD |  Gen 0 | Gen 1 | Gen 2 | Bytes Allocated/Op |
---------------------- |---------- |---------- |------- |---------- |------- |------ |------ |------------------- |
               Default | 5.0061 us | 0.0359 us |   1.00 |      0.00 | 610.00 |     - |     - |             890.47 |
                Pr1311 | 4.9428 us | 0.0277 us |   0.99 |      0.01 | 610.20 |     - |     - |             888.82 |
 EnumerableViewLocator | 5.3197 us | 0.1377 us |   1.06 |      0.03 | 687.45 |     - |     - |           1,000.36 |
```

Where Pr1311 is (solid naming, I know) the functional changes, without IEnumerable, and the Enumerable test is with wrapping into an IEnumerable.

```
 public interface IFooViewModel { }

        public class FooViewModel : ReactiveObject, IFooViewModel { }

        public interface IFooView : IViewFor<IFooViewModel> { }

        public class FooView : IFooView
        {
            object IViewFor.ViewModel
            {
                get { return ViewModel; }
                set { ViewModel = (IFooViewModel)value; }
            }
            public IFooViewModel ViewModel { get; set; }
        }

        private readonly ModernDependencyResolver resolver = new ModernDependencyResolver();
        private readonly IDisposable resolverDisposable;
        private readonly DefaultViewLocator defaultViewLocator = new DefaultViewLocator();
        private readonly PR1311ViewLocator pr1311ViewLocator = new PR1311ViewLocator();
        private readonly EnumerableViewLocator enumerableViewLocator = new EnumerableViewLocator();
        private readonly FooViewModel viewModel = new FooViewModel();

        public ViewResolutionBenchmark()
        {
            resolver.InitializeSplat();
            resolver.InitializeReactiveUI();
            resolver.Register(() => new FooView(), typeof(IViewFor<FooViewModel>));
            resolverDisposable = resolver.WithResolver();
        }

        [Benchmark(Baseline = true)]
        public object Default() => defaultViewLocator.ResolveView(viewModel);

        [Benchmark]
        public object Pr1311() => pr1311ViewLocator.ResolveView(viewModel);

        [Benchmark]
        public object EnumerableViewLocator() => enumerableViewLocator.ResolveView(viewModel);

        public void Dispose()
        {
            resolverDisposable.Dispose();
        }
```

We're making more allocations (10%) but I think it'll depend on route through, I can do more benchmarks if wanted, but on the basis of this I'm happy to not do the Enumerable refactoring/